dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/NetworkApi.java

```diff
@@ -0,0 +1,453 @@
+   @Fallback(NullOnNotFoundOr404.class)
+   Response deleteVlan(@PayloadParam("id") String vlanId);
+
+   @Named("network:addPublicIpBlock")
+   @POST
+   @Path("/addPublicIpBlock")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   Response addPublicIpBlock(@PayloadParam("networkDomainId") String networkDomainId);
+
+   @Named("network:publicIpBlock")
+   @GET
+   @Path("/publicIpBlock")
+   @ResponseParser(ParsePublicIpBlocks.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   PaginatedCollection<PublicIpBlock> listPublicIPv4AddressBlocks(@QueryParam("networkDomainId") String networkDomainId,
+         PaginationOptions options);
+
+   @Named("network:publicIpBlock")
+   @GET
+   @Path("/publicIpBlock")
+   @Transform(ParsePublicIpBlocks.ToPagedIterable.class)
+   @ResponseParser(ParsePublicIpBlocks.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<PublicIpBlock> listPublicIPv4AddressBlocks(@QueryParam("networkDomainId") String networkDomainId);
+
+   @Named("network:removePublicIpBlock")
+   @POST
+   @Path("/removePublicIpBlock")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(NullOnNotFoundOr404.class)
+   Response removePublicIpBlock(@PayloadParam("id") String publicIpBlockId);
+
+   @Named("server:getPublicIpBlock")
+   @GET
+   @Path("/publicIpBlock/{id}")
+   @Fallback(NullOnNotFoundOr404.class)
+   PublicIpBlock getPublicIPv4AddressBlock(@PathParam("id") String publicIPv4AddressBlockId);
+
+   @Named("network:createNatRule")
+   @POST
+   @Path("/createNatRule")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   Response createNatRule(@PayloadParam("networkDomainId") String networkDomainId,
+         @PayloadParam("internalIp") String internalIp, @PayloadParam("externalIp") String externalIp);
+
+   @Named("network:natRule")
+   @GET
+   @Path("/natRule")
+   @ResponseParser(ParseNatRules.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   PaginatedCollection<NatRule> listNatRules(@QueryParam("networkDomainId") String networkDomainId,
+         PaginationOptions options);
+
+   @Named("network:natRule")
+   @GET
+   @Path("/natRule")
+   @Transform(ParseNatRules.ToPagedIterable.class)
+   @ResponseParser(ParseNatRules.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<NatRule> listNatRules(@QueryParam("networkDomainId") String networkDomainId);
+
+   @Named("server:getNatRule")
+   @GET
+   @Path("/natRule/{id}")
+   @Fallback(NullOnNotFoundOr404.class)
+   NatRule getNatRule(@PathParam("id") String natRuleId);
+
+   @Named("network:deleteNatRule")
+   @POST
+   @Path("/deleteNatRule")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   Response deleteNatRule(@PayloadParam("id") String natRuleId);
```

Comment:
generally jclouds uses @Fallback(VoidOnNotFoundOr404.class) and uses void delete*(...)
do you think Response is more useful here?